.github/workflows/ssh-runner.yml

```diff
@@ -0,0 +1,59 @@
+name: SSH into our runners
+
+on:
+  workflow_dispatch:
+    inputs:
+      runner_type:
+        description: 'Type of runner to test (a10 or t4)'
+        required: true 
+      docker_image:
+        description: 'Name of the Docker image'
+        required: true
+
+env:
+  IS_GITHUB_CI: "1"
+  HF_HUB_READ_TOKEN: ${{ secrets.HF_HUB_READ_TOKEN }}
+  HF_HOME: /mnt/cache 
+  TRANSFORMERS_IS_CI: yes 
+  OMP_NUM_THREADS: 8 
+  MKL_NUM_THREADS: 8 
+  RUN_SLOW: yes # For gated repositories, we still need to agree to share information on the Hub repo. page in order to get access. # This token is created under the bot `hf-transformers-bot`. 
+  SIGOPT_API_TOKEN: ${{ secrets.SIGOPT_API_TOKEN }} 
+  TF_FORCE_GPU_ALLOW_GROWTH: true 
+  RUN_PT_TF_CROSS_TESTS: 1
+
+jobs:
+  ssh_runner:
+    name: SSH
+    runs-on: [single-gpu, nvidia-gpu, ${{ github.event.inputs.runner_type }}, ci]
+    container:
+      image: ${{ github.event.inputs.runner_type }}
+      options: --gpus all --privileged --ipc host -v /mnt/cache/.cache/huggingface:/mnt/cache/
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      
+      - name: Install locally transformers & other libs
+        run: |
+          apt install sudo
+          sudo -H pip install --upgrade pip
+          sudo -H pip uninstall -y transformers 
+          sudo -H pip install -U -e ".[testing]" 
+          MAX_JOBS=4 pip install flash-attn --no-build-isolation
+          pip install bitsandbytes
+      
+      - name: NVIDIA-SMI
+        run: |
+          nvidia-smi
+      
+      - name: Show installed libraries and their versions
+        run: pip freeze
+      
+      - name: Tailscale # In order to be able to SSH when a test fails
+        uses: huggingface/tailscale-action@v1
+        with:
+          authkey: ${{ secrets.TAILSCALE_SSH_AUTHKEY }}
+          slackChannel: ${{ secrets.SLACK_CIFEEDBACK_CHANNEL }}
+          slackToken: ${{ secrets.SLACK_CIFEEDBACK_BOT_TOKEN }}
+          waitForSSH: true
```

Comment:
@glegendre01 Is it possible to timeout the job with explicit amount of time here?

Reply:
yes it's possible. 
`sshTimeout: 10m for example`

Reply:
Hi @glegendre01  I didn't see this `sshTimeout` in [this file](https://github.com/huggingface/tailscale-action/blob/main/action.yaml) - hope I am looking the right file.

What I have in mind (from my experience with CircleCI), there are 2 interesting timeout:

- timeout the job if no ssh connection being done with X minutes (say 10m)
- timeout the job after Y minutes even if there is connection (say 1 hour)

Probably this/these is/are already done in [this file](https://github.com/huggingface/tailscale-action/blob/main/action.yaml) without having them in the `inputs`? (I don't mean to ask them being in `inputs`, just wondering if such timeout are implemented). I do see `timeout 5m` in that file, but not sure its meaning.

Reply:
i think tha tfile is different, you need to look for the file with the `v1` tag: https://github.com/huggingface/tailscale-action/blob/v1/action.yaml#L59 🤯 for the other points, i will let @glegendre01 answer 🙏

Reply:
Nice @younesbelkada thanks a lot

. I think so far we can add `sshTimeout` then merge the PR.

Reply:
Thanks ! Ok yes, I think we can use the default one (5min), I would love to have a review from @glegendre01 before merging it if possible 🙏

Reply:
Oh yes, there is default value 😅